### ISSUE TITLE
Broken or Flaky test: ExternalCompaction_2_IT

### ISSUE THREAD

**Issue by ctubbsii** [apache/accumulo#2501]
**Test name(s)**
 - org.apache.accumulo.test.compaction.ExternalCompaction_2_IT.testExternalCompactionsSucceedsRunWithTableOffline
 
 **Describe the failure observed**
```
[ERROR] Tests run: 6, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 105.877 s <<< FAILURE! - in org.apache.accumulo.test.compaction.ExternalCompaction_2_IT
[ERROR] org.apache.accumulo.test.compaction.ExternalCompaction_2_IT.testExternalCompactionsSucceedsRunWithTableOffline  Time elapsed: 9.167 s  <<< FAILURE!
java.lang.AssertionError: expected:<STARTED> but was:<SUCCEEDED>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at org.apache.accumulo.test.compaction.ExternalCompactionTestUtils.confirmCompactionRunning(ExternalCompactionTestUtils.java:278)
	at org.apache.accumulo.test.compaction.ExternalCompaction_2_IT.testExternalCompactionsSucceedsRunWithTableOffline(ExternalCompaction_2_IT.java:196)
```

**Testing Environment:**
 - Version of this project: 2.1.0-SNAPSHOT
 - First commit known to fail (or current commit): 4ce9ab25d061a7cb1f79b49746637b91c22c40ec
```
Executing Maven:  -B -f /var/lib/jenkins/workspace/2.1/pom.xml -V verify -Dfailsafe.rerunFailingTestsCount=5
Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d)
Maven home: /var/lib/jenkins/tools/hudson.tasks.Maven_MavenInstallation/Maven_3.8.1
Java version: 11.0.14, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-11-openjdk-11.0.14.0.9-2.fc35.x86_64
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.16.8-200.fc35.x86_64", arch: "amd64", family: "unix"
```

**What have you tried already?**
Jenkins re-ran the test and it passed the second time

**Additional context**
N/A

**Comment by ctubbsii** [comment]
I saw this test fail again, timing out twice, then on the third time, it yielded a similar error to the above, but was a different state, before finally passing on the fourth attempt:

```
[ERROR] org.apache.accumulo.test.compaction.ExternalCompaction_2_IT.testDeleteTableCancelsExternalCompaction  Time elapsed: 4.362 s  <<< FAILURE!
java.lang.AssertionError: expected:<CANCELLED> but was:<SUCCEEDED>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at org.apache.accumulo.test.compaction.ExternalCompactionTestUtils.confirmCompactionCompleted(ExternalCompactionTestUtils.java:304)
	at org.apache.accumulo.test.compaction.ExternalCompaction_2_IT.testDeleteTableCancelsExternalCompaction(ExternalCompaction_2_IT.java:329)
```

It seems that this can sometimes get in a SUCCEEDED state unexpectedly.

**Comment by dlmarion** [comment]
I think these imply that the Compactor process is running faster than the test code for some reason.

**Comment by dlmarion** [comment]
It may also be the case that the Compactor from the previous test was not actually stopped from the call in `tearDown`.

**Comment by ctubbsii** [comment]
> I think these imply that the Compactor process is running faster than the test code for some reason.

That's kinda what I was thinking.

> It may also be the case that the Compactor from the previous test was not actually stopped from the call in `tearDown`.

Wouldn't it be likely running on a different port? Or using a unique ID for the compaction task for each test, if it's reusing the same cluster?

**Comment by dlmarion** [comment]
The test code is not communicating with the Compactor. I think what I can do is have each test use a different compaction queue. Then, if a Compactor is running from a previous test for some reason, it won't pick up the compaction. I'll start on that change now. Is ExternalCompaction2_IT the only one currently failing?

**Comment by ctubbsii** [comment]
> Is ExternalCompaction2_IT the only one currently failing

It's the only one I saw to motivate this issue, but it wouldn't hurt to make all of them use unique queue names, like we do generally with unique table names in all our ITs.

**Comment by dlmarion** [comment]
@ctubbsii  - I think this might be resolved with #2507. Are you still seeing issues with the ExternalCompaction ITs?

**Comment by ctubbsii** [comment]
> I think this might be resolved with #2507. Are you still seeing issues with the ExternalCompaction ITs?

I think you are right. I haven't seen it fail once since this change. My bet would be the distinct queue names did the trick. They seem to be very reliable now.
